src/IIIFPresentation/API/Features/Storage/Requests/UpsertCollection.cs

```diff
@@ -77,14 +81,24 @@ public async Task<ModifyEntityResult<PresentationCollection, ModifyCollectionTyp
                 IsPublic = request.Collection.Behavior.IsPublic(),
                 IsStorageCollection = request.Collection.Behavior.IsStorageCollection(),
                 Label = request.Collection.Label,
-                Parent = parentCollection.Id,
-                Slug = request.Collection.Slug,
-                Thumbnail = request.Collection.Thumbnail,
+                Thumbnail = request.Collection.PresentationThumbnail,
                 Tags = request.Collection.Tags,
-                ItemsOrder = request.Collection.ItemsOrder
+            };
+            
+            hierarchy = new Hierarchy
```

Comment:
fwiw, I think you should be able to add this hierarchy to the `databaseCollection.Hierarchies` collection and it should get `CollectionId` and `CustomerId` assigned automatically, granted not much of a saving as it's only a couple of fields